src/action/common/configure_init_service.rs

```diff
@@ -167,35 +182,60 @@ impl Action for ConfigureInitService {
 
     #[tracing::instrument(level = "debug", skip_all)]
     async fn execute(&mut self) -> Result<(), ActionError> {
-        let Self { init, start_daemon } = self;
+        let Self {
+            init,
+            nix_enterprise,
+            start_daemon,
+        } = self;
 
         match init {
             #[cfg(target_os = "macos")]
             InitSystem::Launchd => {
-                let src = std::path::Path::new(DARWIN_NIX_DAEMON_SOURCE);
-                tokio::fs::copy(src, DARWIN_NIX_DAEMON_DEST)
-                    .await
-                    .map_err(|e| {
+                let daemon_file: &str;
+                let domain = "system";
+                let service: &str;
+
+                if *nix_enterprise {
```

Comment:
Would be nice to move this into its own module too; just need to factor out the common stuff and have a "configure_enterprise_init_service" module that just does the enterprise-specific stuff here.